### ISSUE TITLE
Button "Invisible" loses accent `color` when using a `leadingVisual`

### ISSUE THREAD

**Issue by maximedegreve** [primer/react#4333]
### Description

Button variant "invisible" loses accent `color` when using a `leadingVisual`.

https://github.com/primer/react/assets/980622/46ca0aa4-459e-4e16-8138-946879a46bf2

In Figma the component doesn't lose the accent color but the icon doesn't have the accent color applied when added.

![Screenshot of the Figma problem](https://github.com/primer/react/assets/980622/a02be941-55a9-4b9e-9dce-9f18a6644783)

### Steps to reproduce

[Go to Storybook](https://primer.style/react/storybook/?path=/story/components-button--playground&args=variant:invisible;leadingVisual:EyeIcon)

### Version

Latest

### Browser

Safari

**Comment by mohanadkandil** [comment]
Hey, 
I can work on this, could you please assign it this to me ?

**Comment by TylerJDev** [comment]
Was taking a look at this issue, and I'm a bit unsure of what the expected color should be 🤔 I took a look at the [most recent version of PRC (pre-release)](https://primer-6674213ee4-13348165.drafts.github.io/storybook/?path=/story/components-button--playground&args=variant:invisible), and I see that the default color for "invisible" without the leading visual has changed from the screenshot. I _think_ this was part of https://github.com/primer/react/pull/4940, so I'm not sure if this is the expectation or not for `invisible`. If it is, then I think we can close this issue. 

@langermank, you might have more context than I do 🙇 

<details>
<summary>Screen capture of "invisible" variant in Storybook</summary>

![Screen capture of "invisible" button variant in Storybook, the button has a invisible background with black text saying "button"](https://github.com/user-attachments/assets/4ebed29d-8006-48be-953e-0ff27d410509)

</details>

**Comment by langermank** [comment]
@TylerJDev thanks for tagging me! Yes, the original design for the invisible variant was never intended to retain the accent blue color if visuals were present. Therefore, it was only ever blue if no visuals were present. 

We recently made a change to make the invisible button the default text color in all scenarios. We found that in most cases the blue color was being overridden by consumers, and it was difficult to use the variant with an inconsistent text color. 

As far as I'm concerned there are no current color bugs with Button 😀
